src/challenges/challengeOne/3.Table/Table.test.tsx

```diff
@@ -1,8 +1,33 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Table } from "./Table";
-
-// describe("Table", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { Table } from "./Table";
+
+describe("Table", () => {
+  const columns = [
+    { property: "name", label: "Name" },
+    { property: "email", label: "Email" },
+  ];
+
+  const data = [
+    { id: 1, name: "John Doe", email: "johndoe@example.com" },
+    { id: 2, name: "Jane Smith", email: "janesmith@example.com" },
+  ];
+
+  it("renders table with correct columns and data", () => {
+    render(<Table columns={columns} data={data} />);
+
+    // Check table header
+    columns.forEach((column) => {
+      const columnHeader = screen.getByText(column.label);
+      expect(columnHeader).toBeInTheDocument();
+    });
+
+    // Check table rows and cells
+    data.forEach((item) => {
```

Comment:
We never want logic in our tests, it makes our test reliant on our logic to work. It's always better to simply test the values we expect the user to see.

So we test each value directly
```
expect(screen.getByText('Name')).toBeInTheDocument();
expect(screen.getByText('Email')).toBeInTheDocument();
```

or if you want to use the mocks, that's perfectly fine. 
```
expect(screen.getByText(mockColumnHeaders[0].label)).toBeInTheDocument();
expect(screen.getByText(mockColumnHeaders[1].label)).toBeInTheDocument();
```